src/main/java/service/Processor.java

```diff
@@ -1,47 +1,47 @@
 package service;
 
-import entity.Car;
+import entity.*;
 
 import java.util.Arrays;
 import java.util.NoSuchElementException;
 import java.util.Optional;
 
-public class Processor {
+public class Processor implements  IProcessor{
 
-    private Car[] cars;
+    private ICar[] cars;
 
     public Processor() {
-        cars = new Car[0];
+        cars = new ICar[0];
     }
 
-    public Processor(Car[] cars) {
+    public Processor(ICar[] cars) {
         this.cars = Optional.of(cars)
                 .orElseThrow(() -> new IllegalArgumentException("Car must be not null!"));
     }
 
-    public Car findCarByModel(Car[] cars, String model) {
-       return Arrays.stream(cars)
+    public ICar findCarByModel(ICar[] cars, String model) {
+        return Arrays.stream(cars)
                 .filter(car -> car.getCarModel().name().equalsIgnoreCase(model))
                 .findFirst()
                 .orElseThrow(NoSuchElementException::new);
     }
 
-    public Car findCarByModel(String model) {
-         return findCarByModel(cars, model);
-      }
+    public ICar findCarByModel(String model) {
+        return findCarByModel(cars, model);
+    }
 
-    public void findCarByModelAndYear(Car[] cars, String model, int year) {
+    public void findCarByModelAndYear(ICar[] cars, String model, int year) {
         Arrays.stream(cars)
-                .filter(car -> car.getCarModel().getModel().equals(model))
-                .filter(car -> car.getProductionYear() == year)
+                .filter(car -> car.getCarModel().getModel().equalsIgnoreCase(model))
+                .filter(car -> (2019 - car.getProductionYear()) > year)
```

Comment:
Your method has name "findCarByModelAndYear" but it doesn't return the result and it doesn't print any car according the method's name because you have a mistake in calculation. 
Your calculation if I put send 2014 as year to your method:
(2019 - 2014 > 2014) == 5 > 2014 == false;
For getting true I have to send age of the care not year:
(2019 - 2014 > 4) == 5 > 4 == true;